### PR TITLE
Operations/measurements on all wires are drawn correctly with mid-circuit measurements

### PR DESCRIPTION
**Pull request by mudit2812**
**Context:**
Bug fix for #5499 

**Description of the Change:**
Adding labels to layer strings only extends to wires, not classical bits.

**Benefits:**
Drawer works correctly with mid-circuit measurements

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

File: doc/releases/changelog-dev.md [hunk 1]
```diff
@@ -333,6 +333,8 @@
 
 <h3>Bug fixes 🐛</h3>
 
+* Operators applied to all wires are now drawn correctly in a circuit with mid-circuit measurements.
```
Comment (mudit2812) [inline doc/releases/changelog-dev.md:336]:
```suggestion
* Operators applied to all wires are now drawn correctly in a circuit with mid-circuit measurements.
  [(#5501)](https://github.com/PennyLaneAI/pennylane/pull/5501)
```

**Comment by mudit2812** [comment]
[sc-61127]

File: pennylane/drawer/tape_text.py [hunk 2]
```diff
@@ -225,7 +226,8 @@ def _add_measurement(m, layer_str, config):
         meas_label = m.return_type.value
 
     if len(m.wires) == 0:  # state or probability across all wires
-        for i, s in enumerate(layer_str):
+        n_wires = len(config.wire_map)
+        for i, s in enumerate(layer_str[:n_wires]):
```
Comment (dwierichs) [inline pennylane/drawer/tape_text.py:230]: Excellent catch that this happens to measurements as well! :100:

File: tests/drawer/test_draw.py [hunk 3]
```diff
@@ -329,6 +331,51 @@ def func():
...
+            qml.cond(m, qml.X)(0)
+            qml.apply(op)
+            return qml.expval(qml.Z(0))
+
+        # Stripping to remove trailing white-space because length of white-space at the
+        # end of the drawing depends on the length of each individual line
+        drawing = qml.draw(func, decimals=decimals)().strip()
+        label = op.label(decimals=decimals).replace("\n", "")
+        expected_drawing = (
+            f"0: ──X──┤↗├──X──{label}─┤  <Z>\n1: ──X───║───║──{label}─┤     \n         ╚═══╝"
```
Comment (dwierichs) [inline tests/drawer/test_draw.py:354]: One _could_ do this for (partial) human readability and line length (black won't break these strings, afaik?)
Consider it a very weak suggestion :sweat_smile: 

```suggestion
            f"0: ──X──┤↗├──X──{label}─┤  <Z>\n"
            f"1: ──X───║───║──{label}─┤     \n"
            "         ╚═══╝"
```

**Review by dwierichs — APPROVED** [review]
Nice, thanks for the quick fix @mudit2812 
And nice that you caught the analogous measurement problem, too!
Only have one weak suggestion

**Review by albi3ro — APPROVED** [review]

File: tests/drawer/test_draw.py (hunk 3 above)
Comment (mudit2812) [reply tests/drawer/test_draw.py:354]: I did have it that way initially but black kept reformatting it :(

Comment (dwierichs) [reply tests/drawer/test_draw.py:354]: Fair enough, then it's not supposed to be :D

**Pull request merged**